zerver/webhooks/zapier/view.py

```diff
@@ -4,46 +4,16 @@
 from django.utils.translation import ugettext as _
 
 from zerver.decorator import api_key_only_webhook_view
-from zerver.lib.actions import check_send_private_message_from_emails
 from zerver.lib.request import REQ, has_request_variables
 from zerver.lib.response import json_error, json_success
-from zerver.lib.webhooks.common import check_send_webhook_message, \
-    validate_extract_webhook_http_header
+from zerver.lib.webhooks.common import check_send_webhook_message
 from zerver.models import UserProfile
 
 @api_key_only_webhook_view('Zapier', notify_bot_owner_on_invalid_json=False)
 @has_request_variables
 def api_zapier_webhook(request: HttpRequest, user_profile: UserProfile,
                        payload: Dict[str, Any]=REQ(argument_type='body')) -> HttpResponse:
 
-    # A request with the ZapierZulipApp user agent is a request from
-    # the official Zulip app for Zapier
-    user_agent = validate_extract_webhook_http_header(
-        request, 'USER_AGENT', 'Zapier', fatal=False)
-    if user_agent == 'ZapierZulipApp':
-        event_type = payload.get('type')
-        if event_type == 'auth':
-            # The bot's details are used by Zapier to format a connection
-            # label for users to be able to distinguish between different
-            # Zulip bots and API keys in their UI
-            return json_success({
-                'bot_name': user_profile.full_name,
-                'bot_email': user_profile.email,
-                'bot_id': user_profile.id
-            })
```

Comment:
I think we might want to keep the `auth` endpoint, since at the very least it's a way to confirm if the Zulip server is a new enough version to support this feature.

Reply:
Though I think it makes sense to release 2.0 without this code, because having it there in 2.0 will undermine our ability to do the check I just described.  So I'm merging this commit.

Reply:
@timabbott: Well, first of all, we now use `zulip-js's` `getProfile` function to do the initial auth testing, see [here](https://github.com/zulip/zulip-js/blob/master/src/resources/messages.js#L15). The reason I completely switched over to `zulip-js` was that it didn't make any sense to have a discrepancy between what kind of auth we do initially vs. what kind of auth is done for sending messages later. That seems like terrible design because the user could potentially authenticate successfully initially through the webhook endpoint, only to find out that the same API key doesn't work when the code for sending messages can't use it in Basic auth. While I am not sure how many users would actually make that mistake, it still didn't seem wise to let it happen anyway.